src/c/mip/mip_parser.c

```diff
@@ -342,8 +346,20 @@ void mip_parser_parse(mip_parser* parser, const uint8_t* input_buffer, size_t in
                 MIP_DIAG_INC(parser->_diag_valid_packets, 1);
                 MIP_DIAG_INC(parser->_diag_packet_bytes, expected_packet_length);
 
+                bool stop = true;
+
+                // If there's no callback, assume the user just wants one packet at a time.
                 if(parser->_callback)
```

Comment:
You could remove the stop bool and just check if (!parser->_callback || !parser->_callback(parser->_callback_object, &packet, parser->_start_time))